src/Model/MyParcelCustomsItem.php

```diff
@@ -48,14 +54,23 @@ public function getDescription()
      * Required: Yes
      *
      * @param mixed $description
+     * @param string|int|AbstractCarrier $carrier
      * @return $this
+     *
+     * @throws \Exception
      */
-    public function setDescription($description)
+    public function setDescription($description, $carrier = null): self
     {
-        /**
-         * Description cut after 47 chars
-         */
-        $this->description = Str::limit($description, self::DESCRIPTION_MAX_LENGTH);
+        $maxLength = AbstractConsignment::DESCRIPTION_MAX_LENGTH;
+
+        if ($carrier) {
+            $carrier   = ConsignmentFactory::createFromCarrier($carrier);
+            $maxLength = $carrier::DESCRIPTION_MAX_LENGTH;
+        }
+
+        $description = Str::limit($description, $maxLength - 3, '...');
+
+        $this->description = $description;
```

Comment:
het is mogelijk dat description hier een int is, misschien nog even casten naar een string in deze regel zodat we nooit (meer) foute waarden in description kunnen krijgen